### PR TITLE
Add hoverinfo skip

### PR DESCRIPTION
**Pull request by john-soklaski**
Add a new hoverinfo flag 'skip'. This prevents a trace from being considered for hovering or clicking. Addresses #678.

### CODE REVIEW

**Comment by john-soklaski** [comment]
@etpinard I'm working on testing gl2d and gl3d plots

**Comment by john-soklaski** [comment]
@etpinard I was able to add some support for gl2d and gl3d. 

There are a few limitations that I'm aware of: because gl2d and gl3d use the gl-plot2d and gl-plot3d packages for picking, traces with `hoverinfo: 'skip'` still affect the picking logic. I was able to prevent the spikes from being shown for the gl2d plot, but gl-plot3d handles the spikes drawing internally.

**Comment by etpinard** [comment]
> I was able to prevent the spikes from being shown for the gl2d plot, but gl-plot3d handles the spikes drawing internally.

Spikes in gl3d plots are controlled via the `layout.scene.?axis.showspikes` options. They are effectively decoupled from the hover logic. See how the first modebar button from the right toggles `showspikes` [here](https://github.com/plotly/plotly.js/blob/acff0da9f9574ef8ded7a1aa48817c6d8d1a571c/src/components/modebar/buttons.js#L348). In brief, you don't have to worry about the 3D spikes. 

Regarding gl3d and gl2d, I was most concerned about making sure that `'skip'` does _as good as_ `'none'` at this stage. I don't mind if `'skip'` only _works_ for plots that pass through `Fx.hover`.

Great. So looks like you got everything right. Thanks for adding those tests!

The only thing missing is adding information about `'skip'` vs `'none'` in the `hoverinfo` attribute declaration [description](https://github.com/plotly/plotly.js/pull/851/files#diff-bebc33badeef34dcf038b61fc863d3e6R77) field.

I'll take this PR for a final spin on Monday, before merging. Thanks again for you efforts.

**Comment by john-soklaski** [comment]
@etpinard Thanks for taking a look, I've added some information to the description about the behavior of `skip` and `none`.

File: src/plots/attributes.js [hunk 1]
```diff
@@ -74,7 +74,11 @@ module.exports = {
         flags: ['x', 'y', 'z', 'text', 'name'],
         extras: ['all', 'none', 'skip'],
         dflt: 'all',
-        description: 'Determines which trace information appear on hover.'
+        description: [
+            'Determines which trace information appear on hover.',
+            'If `none` or `skip` are set, no information is displayed upon hovering.',
+            'But, if `none` is set, click and hover events are still fired.'
```
Comment (etpinard) [inline src/plots/attributes.js]: Great. Thanks!

**Comment by etpinard** [comment]
Great PR. @john-soklaski thanks very much for your efforts :tada:

**Pull request merged**